use_cases/fetcher.go

```diff
@@ -19,26 +21,65 @@ type Fetcher struct {
 	storage writer
 }
 
+type result struct {
+	Error   error
+	Pokemon models.Pokemon
+}
+
 func NewFetcher(api api, storage writer) Fetcher {
 	return Fetcher{api, storage}
 }
 
 func (f Fetcher) Fetch(from, to int) error {
 	var pokemons []models.Pokemon
-	for id := from; id <= to; id++ {
-		pokemon, err := f.api.FetchPokemon(id)
-		if err != nil {
-			return err
-		}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-		var flatAbilities []string
-		for _, t := range pokemon.Abilities {
-			flatAbilities = append(flatAbilities, t.Ability.URL)
+	ch := f.generator(ctx, from, to)
+	for result := range ch {
```

Comment:
This is not a generator pattern as you are blocking your code, waiting for a certain response in the channel.